fix: bank selection filter remaining bug

tkjftkjf opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **새 기능**
  * 은행 선택 UI가 별도 위젯 흐름으로 재구성되어 은행 목록, 검색 입력, 상세 뷰가 독립적으로 동작합니다.
  * 목록과 상세 전환이 애니메이션으로 매끄럽게 처리되며, 선택은 콜백으로 상위에 전달됩니다.

* **버그 수정**
  * 검색 상태와 스크롤 위치 관리가 안정화되어 동작 일관성이 개선되었습니다.
  * 은행 항목 선택 시 불필요한 로컬 상태 변경을 제거해 신뢰도가 향상되었습니다.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tkjftkjf committed:
fix: fix bank selection filter remaining bug

tkjftkjf committed:
feat: seperate BankList Widget

tkjftkjf committed:
fix: implement scroll listener to BankList for tracking scroll position

tkjftkjf committed:
fix: minor changes

- remove temporary solution
- change onSearchValueChanged from nullable to non-null

tkjftkjf committed:
refactor: change BankList widget to private

tkjftkjf committed:
fix: remove unused key

2paperstar commented on lib/app/modules/user/presentation/pages/account_number_settings_page.dart:
```diff
@@ -236,7 +292,8 @@ class _SelectBankDialogState extends State<_SelectBankDialog> {
         const SizedBox(height: 20),
         PotTextField(
-          onChanged: (value) => setState(() => _search = value),
+          onChanged: widget
```
코멘트 지워주세요

2paperstar replied:
그리고 이 함수는 필요 없습니다 (상단 코멘트 참조)

2paperstar commented on lib/app/modules/user/presentation/pages/account_number_settings_page.dart:
```diff
@@ -188,10 +188,83 @@ class _SelectBankDialog extends StatefulWidget {
+                search: _search,
+                onSearchValueChanged: (value) {
+                  setState(() => _search = value);
+                },
```
_serach도 BankList 내부의 state로 만들 수 있겠죠?

tkjftkjf committed:
refactor: remvoe unnecessary coupling and match name styles

- _search를 BankList에서만 이용하도록 변경하였음
- onBankSelected method를 스타일에 맞게 onSelectBank로 변경
- BankListState내부 필드들을 private으로 변경

tkjftkjf committed:
chore: remove comments

2paperstar commented on lib/app/modules/user/presentation/pages/account_number_settings_page.dart:
```diff
@@ -188,9 +188,71 @@ class _SelectBankDialog extends StatefulWidget {
+  }
+}
+
+class _Bank extends StatelessWidget {
+  const _Bank({required this.bank});
+
+  final BankEntity bank;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Container(
+          padding: EdgeInsets.all(8),
+          height: 64,
+          width: 64,
+          decoration: BoxDecoration(
+            border: Border.all(color: Palette.borderGrey),
+            borderRadius: BorderRadius.circular(8),
+          ),
+          child: Image.network(bank.logoUrl),
+        ),
+        const SizedBox(height: 4),
+        Text(bank.name, style: TextStyles.description),
+      ],
+    );
+  }
+}
```
#422 에서 계속
